src/app/invitations/invitation-view/invitation-view.component.ts

```diff
@@ -12,17 +21,171 @@ import { faWarning } from '@fortawesome/free-solid-svg-icons';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class InvitationViewComponent implements OnInit {
-  invitation$?: Observable<Invitation>;
+  payload$?: Observable<Payload>;
 
   faWarning = faWarning;
+  faInfo = faInfo;
 
-  constructor(private activatedRoute: ActivatedRoute, private invitationService: InvitationCollectionService) {}
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private invitationService: InvitationCollectionService,
+    private invitationRedeemRequestService: InvitationRedeemRequestCollectionService,
+    private messageService: MessageService,
+    private router: Router,
+    private http: HttpClient,
+    private organizationService: OrganizationCollectionService,
+    private billingService: BillingEntityCollectionService,
+    private teamService: TeamCollectionService,
+    private urlGenerator: KubernetesUrlGenerator
+  ) {}
 
   ngOnInit(): void {
-    const name = this.activatedRoute.snapshot.paramMap.get('name') ?? '';
-    if (!name) {
+    const invitationName = this.activatedRoute.snapshot.paramMap.get('name')?.trim() ?? '';
+    if (!invitationName) {
       throw new Error('name is required as path parameter in URL');
     }
-    this.invitation$ = this.invitationService.getByKeyMemoized(name);
+
+    const tokenInQuery = this.activatedRoute.snapshot.queryParamMap.get('token');
+    const tokenFromStorage = window.localStorage.getItem(invitationTokenLocalStorageKey);
```

Comment:
What if the user has local storage disabled? Especially on safari, localStorage isn't always reliably there. I would recommend handling any errors if localStorage cannot be accessed, and show the user a message that localStorage is required for this feature